umpost/um2netcdf.py

```diff
@@ -99,46 +126,193 @@ def convert_proleptic(time):
-        lon.var_name = 'lon_river'
-    elif (lon.points[0] == 0 and grid_type == 'EG') or \
-         (np.allclose(0.5*dlon, lon.points[0]) and grid_type == 'ND'):
-        lon.var_name = 'lon_u'
+        return False  # TODO: Is this situation ever valid?
+
+
+def is_lon_u_grid(longitude_points, grid_type, dlon):
+    """
+    Check whether longitude points are on the lon_u grid.
+
+    Parameters
+    ----------
+    longitude_points: (array) 1D array of longitude grid points.
+    grid_type: (string) model horizontal grid type.
+    dlon: (float) zonal spacing between longitude grid points.
+    """
+    min_longitude = longitude_points[0]
+    min_lon_u_nd_grid = 0.5*dlon
+    min_lon_u_eg_grid = 0
+
+    if grid_type == GRID_END_GAME:
+        return min_longitude == min_lon_u_eg_grid
+    elif grid_type == GRID_NEW_DYNAMICS:
+        return np.allclose(min_lon_u_nd_grid, min_longitude)
     else:
-        lon.var_name = 'lon'
+        return False  # TODO: Is this situation ever valid?
+
+
+def add_latlon_coord_bounds(cube_coordinate):
+    """
+    Add bounds to horizontal coordinate (longitude or latitude) if
+    they don't already exist. Edits coordinate in place.
+
+    Parameters
+    ----------
+    cube_coordinate: coordinate object from iris cube.
+    """
+    coordinate_name = cube_coordinate.name()
+    if coordinate_name not in [LONGITUDE, LATITUDE]:
+        raise ValueError(
+                f"Wrong coordinate {coordinate_name} supplied. "
+                f"Expected one of {LONGITUDE}, {LATITUDE}."
+            )
+
+    # Only add bounds if not already present.
+    if not cube_coordinate.has_bounds():
+        if len(cube_coordinate.points) > 1:
+            cube_coordinate.guess_bounds()
+        else:
+            # For length 1, assume it's global.
+            # guess_bounds doesn't work in this case.
+            cube_coordinate.bounds = GLOBAL_COORD_BOUNDS[coordinate_name]
+
+
+def fix_latlon_coords(cube, grid_type, dlat, dlon):
+    """
+    Wrapper function to modify cube's horizontal coordinates
+    (latitude and longitude). Converts to float64, adds grid bounds,
+    and renames coordinates. Modifies cube in place.
+
+    NB - grid spacings dlat and dlon only refer to variables on the main
+    horizontal grids, and not the river grid.
+
+    Parameters
+    ----------
+    cube: Iris cube object (modified in place).
+    grid_type: (string) model horizontal grid type.
+    dlat: (float) meridional spacing between latitude grid points.
+    dlon: (float) zonal spacing between longitude grid points.
+    """
+
+    try:
+        latitude_coordinate = cube.coord(LATITUDE)
+        longitude_coordinate = cube.coord(LONGITUDE)
+    except iris.exceptions.CoordinateNotFoundError as CoordError:
+        msg = (
+            "Missing latitude or longitude coordinate for variable (possible timeseries?): \n"
+            f"{cube}\n"
+        )
+        raise UnsupportedTimeSeriesError(msg) from CoordError
```

Comment:
There's another styling thing here with the capitalised var name thanks to my earlier comment. I parsed that as a type, not a variable. How is `coord_error` instead?

Reply:
Yeah good idea, fixed in 6a4c876